s3-source-connector/src/test/java/io/aiven/kafka/connect/s3/source/input/JsonTransformerTest.java

```diff
@@ -63,28 +65,34 @@ void testConfigureValueConverter() {
     void testHandleValueDataWithValidJson() {
         final InputStream validJsonInputStream = new ByteArrayInputStream(
                 "{\"key\":\"value\"}".getBytes(StandardCharsets.UTF_8));
-        final List<Object> jsonNodes = jsonTransformer.getRecords(validJsonInputStream, "testtopic", 1, s3SourceConfig);
+        final IOSupplier<InputStream> inputStreamIOSupplier = () -> validJsonInputStream;
+        final Stream<Object> jsonNodes = jsonTransformer.getRecords(inputStreamIOSupplier, "testtopic", 1,
+                s3SourceConfig);
 
-        assertThat(jsonNodes.size()).isEqualTo(1);
+        assertThat(jsonNodes.collect(Collectors.toList())).hasSize(1);
     }
 
     @Test
     void testHandleValueDataWithInvalidJson() {
         final InputStream invalidJsonInputStream = new ByteArrayInputStream(
                 "invalid-json".getBytes(StandardCharsets.UTF_8));
+        final IOSupplier<InputStream> inputStreamIOSupplier = () -> invalidJsonInputStream;
 
-        final List<Object> jsonNodes = jsonTransformer.getRecords(invalidJsonInputStream, "testtopic", 1,
+        final Stream<Object> jsonNodes = jsonTransformer.getRecords(inputStreamIOSupplier, "testtopic", 1,
                 s3SourceConfig);
 
-        assertThat(jsonNodes.size()).isEqualTo(0);
+        assertThat(jsonNodes.collect(Collectors.toList())).hasSize(0);
     }
 
```

Comment:
It would be great to get a test that tests the behaviour of the IOException thrown by the readJsonRecordsAsStream for the captured and logged entry and also for the Exception caught and thrown by getRecords so if that behaviour ever changes the unit tests will break and they have to be addressed as well.

Reply:
For logged entry assertion, we didn't have any package already. Need to add something like nl.altindag.log.LogCaptor to dependencies. Will ignore this assertion.

Reply:
Yeah, I don't think we need to worry about capturing the logs, I've done that before for tests but found it flaky, I think more so along the lines, of we expect this exception to be caught and not to have an impact, so if someone comes along later and decide to rethrow we can see the change in behaviour in the tests breaking